test/jdk/java/security/CodeSource/Implies.java

```diff
@@ -48,6 +49,40 @@ public static void main(String[] args) throws Exception {
         // port check should match default port of thatURL
         testImplies(thisURL, thatURL, true);
 
+        thisURL = new URL("http", "204.160.241.0", "file");
+        thatURL = new URL("http", "localhost", "file");
+        // ip address should not imply localhost's IP address
+        testImplies(thisURL, thatURL, false);
+
+        thisURL = new URL("http", "204.160.241.0", "file");
+        thatURL = new URL("http", "*.example.com", "file");
+        // ip address should not imply wildcarded host
+        testImplies(thisURL, thatURL, false);
+
+        InetAddress ia = InetAddress.getLocalHost();
+        thisURL = new URL("http", ia.getHostAddress(), "file");
+        thatURL = new URL("http", ia.getHostName(), "file");
+        // ip address should imply host name with same ip address
```

Comment:
The specification speaks about IP addresses, specifically:

> If this object was initialized with a single IP address and one of <i>p</i>'s IP addresses is equal to this object's IP address.

So is this test for implying a IP address against a host name valid?

Reply:
Yes, it is checking that [the locations specified as ip address and hostname are the same](https://github.com/openjdk/jdk/blob/5ed72775a775d6c96be6de7d9540dc0a80cdd67b/src/java.base/share/classes/java/security/CodeSource.java#L398). This isn't testing any of the `SocketPermission` rules, but I thought it would be useful to add it as a test as I didn't find any existing `CodeSource` tests that checked for that condition.

Reply:
I see. Looks fine to me then. I hadn't paid attention that this behaviour is from the `CodeSource's` existing specification rather than from what it was borrowing from `SocketPermission`.